helsingborg-stad/app-mitt-helsingborg
# Feat/fetch application version status redo

## Explain the changes you’ve made
Added new service and context for fetching application version status and provide `isCompatible` state to the application. The value is telling if the application version needs to be updated or not.

If the application version is compatible, the user can sign in, otherwise the user is redirected to android or ios app store.

Also added tests (including new module) for hooks.

## Explain why these changes are made
We want to block users for sining in to the application if their device has an old version of the Mitt Helsingborg application which isn't compatible with the current running backend and/or get new required features.

## Explain your solution
Added a new file, `ApplicationVersionService.ts` which fetches the version status.

Added a new context, AppCompatibilityContext
- that consults ApplicationVersionService
- is used by AuthContext to force logout if app is incompatible and sets an error message on signing screen.

## How to test

1. Checkout this branch
2. Fire upp the simulator by running the command `yarn ios` or `yarn android`
3. Make sure you have the version-api deployed and versionEnvs deployed 
4. Make sure that the value in the versionEnvs looks like the following:
```js
{
  "versions": {
      "ios": {
          "min": "1.1.6",  -> Version of your choosing
          "max": "3.1.0", -> Version of your choosing
          "updateUrl": "https://apps.apple.com/se/app/mitt-helsingborg/id1467736612" // Does not work on ios simulator
      },
      "android": {
          "min": "1.1.4", -> Version of your choosing
          "max": "1.1.6", -> Version of your choosing
          "updateUrl": "https://play.google.com/store/apps/details?id=com.mitthelsingborg&gl=SE"
      }
  }
}
```
Test cases:
1. `current application version` is the same as `max` -> No update required
2. `current application version` is between `min` and`max` -> No update required
3. `current application version` is lesser than `min` -> Update required

## Tested environments

- [] Storybook on a iOS device/simulator.
- [] Building the Application on a Android device/simulator.
